docs/content/Caching/Running-in-Production.md

```diff
@@ -0,0 +1,212 @@
+---
+title: Running in Production
+permalink: /caching/running-in-production
+category: Caching
+menuOrder: 4
+---
+
+Cube.js makes use of two different kinds of cache:
+
+- Redis, for in-memory storage of query results
+- Cube Store for persisting pre-aggregations
+
+Redis is included by default in development mode. In production, we **strongly**
+recommend running Redis as a separate service.
+
+Cube Store is enabled by default when running Cube.js in development mode. In
+production, Cube Store **must** run as a separate process. The easiest way to do
+this is to use the official Docker images for Cube.js and Cube Store.
+
+You can run Cube Store with Docker with the following command:
+
+```bash
+docker run -p 3030:3030 cubejs/cubestore
+```
+
+<!-- prettier-ignore-start -->
+[[info | ]]
+| Cube Store can further be configured via environment variables. To see a
+| complete reference, please consult the [Cube Store section of the Environment
+| Variables reference][ref-config-env].
+<!-- prettier-ignore-end -->
+
+Next, run Cube.js and tell it to connect to Cube Store running on `localhost`
+(on the default port `3030`):
+
+```bash
+docker run -p 4000:4000 \
+  -e CUBEJS_CUBESTORE_HOST=localhost \
+  -v ${PWD}:/cube/conf \
+  cubejs/cube
+```
+
+In the command above, we're specifying `CUBEJS_CUBESTORE_HOST` to let Cube.js
+know where Cube Store is running.
+
+You can also use Docker Compose to achieve the same:
+
+```yaml
+version: '2.2'
+services:
+  cubestore:
+    image: cubejs/cubestore:latest
+    environment:
+      - CUBESTORE_REMOTE_DIR=/cube/data
+    volumes:
+      - .cubestore:/cube/data
+
+  cube:
+    image: cubejs/cube:latest
+    ports:
+      - 4000:4000
+    environment:
+      - CUBEJS_CUBESTORE_HOST=localhost
+      - CUBEJS_CUBESTORE_PORT=3030
+    depends_on:
+      - cubestore
+    links:
+      - cubestore
+    volumes:
+      - ./schema:/cube/conf/schema
+```
+
+### Scaling
+
+<!-- prettier-ignore-start -->
+[[warning | ]]
+| Cube Store _can_ be run in a single instance mode, but this is usually
+| unsuitable for production deployments. For high concurrency and data
+| throughput, we **strongly** recommend running Cube Store as a cluster of
+| multiple instances instead.
+<!-- prettier-ignore-end -->
+
+Scaling Cube Store for a higher concurrency is relatively simple when running in
+cluster mode. Because [the storage layer](#running-in-production-storage) is
+decoupled from the query processing engine, you can horizontally scale your Cube
+Store cluster for as much concurrency as you require.
+
+In cluster mode, Cube Store runs two kinds of nodes:
+
+- a single **router** node handles incoming client connections, manages database
+  metadata and serves simple queries.
+- multiple **worker** nodes which execute SQL queries
+
+In terms of configuration, both the router and worker nodes **must** specify the
```

Comment:
I feel like there's a lot of information (of different env. variables) in this paragraph. Would it be better to organize them in bullet points or even in a table?

Reply:
How about a table?

| Environment Variable | Specify on Router? | Specify on Worker? |
|---|---|---|
| `CUBESTORE_SERVER_NAME` | Yes | Yes |
| `CUBESTORE_META_PORT` | Yes | No |
| `CUBESTORE_WORKERS` | Yes | No |
| `CUBESTORE_WORKER_PORT` | No | Yes |
| `CUBESTORE_META_ADDR` | No | Yes |

Reply:
@hassankhan Looks great! I suggest maybe even changing "No" → "—" to make it visually more distinguishable from "Yes".